tests/capture/test_capture_qnode.py

```diff
@@ -0,0 +1,230 @@
+    expected_kwargs = {"diff_method": "best"}
+    expected_kwargs.update(circuit.execute_kwargs)
+    assert eqn0.params["qnode_kwargs"] == expected_kwargs
+
+    qfunc_jaxpr = eqn0.params["qfunc_jaxpr"]
+    assert qfunc_jaxpr.eqns[0].primitive == qml.RX._primitive
+    assert qfunc_jaxpr.eqns[1].primitive == qml.Z._primitive
+    assert qfunc_jaxpr.eqns[2].primitive == qml.measurements.ExpectationMP._obs_primitive
+
+    assert len(eqn0.outvars) == 1
+    assert eqn0.outvars[0].aval == jax.core.ShapedArray(
+        (), jax.numpy.float64 if x64_mode else jax.numpy.float32
+    )
+
+    output = jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, 0.5)
+    assert qml.math.allclose(output[0], jax.numpy.cos(0.5))
+
+    jax.config.update("jax_enable_x64", initial_mode)
+
+
+@pytest.mark.parametrize("dev_name", ("default.qubit", "default.qubit.legacy"))
+@pytest.mark.parametrize("x64_mode", (True, False))
+def test_overriding_shots(dev_name, x64_mode):
+    """Test that the number of shots can be overridden on call."""
+    initial_mode = jax.config.jax_enable_x64
+    jax.config.update("jax_enable_x64", x64_mode)
+
+    dev = qml.device(dev_name, wires=1)
+
+    @qml.qnode(dev)
+    def circuit():
+        return qml.sample()
+
+    jaxpr = jax.make_jaxpr(partial(circuit, shots=50))()
+    assert len(jaxpr.eqns) == 1
+    eqn0 = jaxpr.eqns[0]
+
+    assert eqn0.primitive == qnode_prim
+    assert eqn0.params["device"] == dev
+    assert eqn0.params["shots"] == qml.measurements.Shots(50)
+    assert (
+        eqn0.params["qfunc_jaxpr"].eqns[0].primitive == qml.measurements.SampleMP._wires_primitive
+    )
+
+    assert eqn0.outvars[0].aval == jax.core.ShapedArray(
+        (50,), jax.numpy.int64 if x64_mode else jax.numpy.int32
+    )
+
+    res = jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts)
+    assert qml.math.allclose(res, jax.numpy.zeros((50,)))
+
+    jax.config.update("jax_enable_x64", initial_mode)
+
+
+def test_providing_keyword_argument():
+    """Test that keyword arguments can be provided to the qnode."""
+
+    @qml.qnode(qml.device("default.qubit", wires=1))
+    def circuit(*, n_iterations=0):
+        for _ in range(n_iterations):
+            qml.X(0)
+        return qml.probs()
+
+    jaxpr = jax.make_jaxpr(partial(circuit, n_iterations=3))()
+
+    assert jaxpr.eqns[0].primitive == qnode_prim
+
+    qfunc_jaxpr = jaxpr.eqns[0].params["qfunc_jaxpr"]
+    for i in range(3):
+        assert qfunc_jaxpr.eqns[i].primitive == qml.PauliX._primitive
+    assert len(qfunc_jaxpr.eqns) == 4
+
+
+@pytest.mark.parametrize("x64_mode", (True, False))
+def test_multiple_measurements(x64_mode):
+    """Test that the qnode can return multiple measurements."""
+    initial_mode = jax.config.jax_enable_x64
+    jax.config.update("jax_enable_x64", x64_mode)
+
+    @qml.qnode(qml.device("default.qubit", wires=3, shots=50))
+    def circuit():
+        return qml.sample(), qml.probs(wires=(0, 1)), qml.state()
```

Comment:
I know this is not the purpose of this test, but `state` measurements are not supported with finite shots, so `eval_jaxpr`ing the JAXPR created in this test would fail. Might be nice to use `shots=None` instead to obtain valid PL code.

Reply:
This makes me wonder: Do we want validation at the compilation level for this type of inputs? I.e. `StateMP._abstract_eval` could raise an error when it is being passed shots :)

Reply:
Good point, since we are doing that validation with `qml.sample()` .  I think I'll leave out the error with shots and state for this PR, but it might be worth adding it in at some point in the future.

I chose sample, probs, and state here since that covered int, float, and complex types.  I'll break out a separate test case for `state` and `density_matrix` to test the complex data types.